### PR TITLE
Brontozaur: Support connecting with existing tcp stream

### PR DESCRIPTION
**Pull request by TheCharlatan**
This supports creating connecting Brontozaur sessions with an external tcp stream, for example one spawned by a socks proxy.

Example usage:

```
use internet2::addr::{InetSocketAddr, LocalNode};
use std::str::FromStr;
use socks::Socks5Stream;
use internet2::session::BrontozaurSession;

dummy = InetSocketAddr::from_str("127.0.0.1:12345").unwrap();
let local_node = LocalNode::new(bitcoin::secp256k1::SECP256K1)
let socks_stream = Socks5Stream::connect(
    "127.0.0.1:9050",
    "fdbzgjl7vtezlzwyx6xipvbczq3cuhvxfnypoljuctnl5mqc4bwrr7ad.onion"
).unwrap();
let mut stream = socks_stream.into_inner();
let session = BrontozaurSession::with(stream, local_node.private_key(), dummy).unwrap();
```

### CODE REVIEW

File: src/session/session.rs [hunk 1]
```diff
@@ -432,6 +444,24 @@ impl<const LEN_SIZE: usize>
         )
     }
 
+    fn with_connect_tcp_encrypted(
```
Comment (dr-orlovsky) [inline src/session/session.rs:447]: Can we re-name this method to `with_tcp_encrypted`? The current name is not very much readable

**Review by dr-orlovsky — APPROVED** [review]
ACK 175024294987fda06d6078aa25eee50847abc201

Thank you!
One minor NIT - if you have time you can address it in a follow-up PR. Otherwise I can do it myself (but this will break your code).

File: src/session/session.rs (hunk 1 above)
Comment (TheCharlatan) [reply src/session/session.rs:447]: I hope this is what you wanted: https://github.com/Internet2-WG/rust-internet2/pull/30 . If not feel free to break my code  :P

Comment (dr-orlovsky) [reply src/session/session.rs:447]: Thank you!

**Pull request merged**